### DISCUSSION TITLE
Extracting the total number of signal and background events when fitting a Z-peak

### DISCUSSION THREAD

**Discussion by ikrommyd** [scikit-hep/iminuit#972] [Q&A]
I'm trying to fit a Z-peak with an extended binned maximum likelihood. My workflow is this:
```python
def integral(
    xe,
    s,
    b,
    beta_left,
    m_left,
    scale_left,
    beta_right,
    m_right,
    scale_right,
    loc,
    beta,
    gamma,
    peak,
):
    s_cdf = crystalball_ex.cdf
    b_cdf = cmsshape.cdf
    return s * n.sum() * s_cdf(
        xe, beta_left, m_left, scale_left, beta_right, m_right, scale_right, loc
    ) + b * n.sum() * b_cdf(xe, beta, gamma, peak)


c = cost.ExtendedBinnedNLL(n, xe, integral)
m = Minuit(
    c,
    s=0.9,
    b=0.1,
    beta_left=1,
    m_left=1.1,
    scale_left=5,
    beta_right=1,
    m_right=1.1,
    scale_right=5,
    loc=91,
    beta=0.03,
    gamma=0.1,
    peak=90,
)
m.limits["s", "b"] = (0, None)
m.strategy = 2
m.hesse()
m.migrad()
```
My data are binned in bins from 50 to 130 GeV.
In the end, I'm getting a fit that looks okay like this:
<img width="553" alt="image" src="https://github.com/scikit-hep/iminuit/assets/82155404/dde23fc9-d280-40fb-9d1c-7a35ac57896b">
but the minimum is not valid and all the minos profiles look very jagged and not smooth at all.
<img width="647" alt="image" src="https://github.com/scikit-hep/iminuit/assets/82155404/b2334642-b808-44a0-90c1-03ff754f7817">
The fit also appears to be extremely senstive to starting values for the parameters. I have two questions regarding this.
1) Are there any general guidelines to do this better or does it just mean that the data are not well explained by the combination of those functions?
2) Is it possible to extract the total number of signal and background events or does the fact that we are only fitting in the a range 50-130 prevent us from doing that since the pdfs are not normalized in that range?

**Comment by HDembinski** [comment]
Sorry, the answer is probably obsolete now, but this is caused by the crystallball pdf. The PDF is poorly designed, and hence hard to fit, iminuit cannot help or hurt here. I always recommend the student's t, and if you need something asymmetric, the non-central student's t distribution.
